internal/provider/relationship_resource.go

```diff
@@ -0,0 +1,229 @@
+				Required:            true,
+				PlanModifiers: []planmodifier.String{
+					stringplanmodifier.RequiresReplace(),
+				},
+			},
+		},
+	}
+}
+
+func (r *RelationshipResource) Configure(ctx context.Context, req resource.ConfigureRequest, resp *resource.ConfigureResponse) {
+	// Prevent panic if the provider has not been configured.
+	if req.ProviderData == nil {
+		return
+	}
+
+	client, ok := req.ProviderData.(*authzed.Client)
+
+	if !ok {
+		resp.Diagnostics.AddError(
+			"Unexpected Resource Configure Type",
+			fmt.Sprintf("Expected *authzed.Client, got: %T. Please report this issue to the provider developers.", req.ProviderData),
+		)
+
+		return
+	}
+
+	r.client = client
+}
+
+func parseRelationship(s string) (*authproto.Relationship, error) {
+	rel := tuple.ParseRel(s)
+	if rel == nil {
+		return nil, fmt.Errorf("invalid relationship string: %s", s)
+	}
+
+	return rel, nil
+}
+
+func (r *RelationshipResource) Create(ctx context.Context, req resource.CreateRequest, resp *resource.CreateResponse) {
+	var data RelationsipResourceModel
+
+	// Read Terraform plan data into the model
+	resp.Diagnostics.Append(req.Plan.Get(ctx, &data)...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+	rel, err := parseRelationship(data.Relationship.ValueString())
+	if err != nil {
+		resp.Diagnostics.AddError("Invalid Relationship", fmt.Sprintf("Unable to parse relationship: %s", err))
+		return
+	}
+
+	_, err = r.client.WriteRelationships(ctx, &authproto.WriteRelationshipsRequest{
+		Updates: []*authproto.RelationshipUpdate{
+			{
+				Operation:    authproto.RelationshipUpdate_OPERATION_CREATE,
+				Relationship: rel,
+			},
+		},
+	})
+
+	if err != nil {
+		resp.Diagnostics.AddError("SpiceDB Client Error", fmt.Sprintf("Unable to create relationships, got error: %s", err))
+		return
+	}
+
+	tflog.Trace(ctx, "Created SpiceDB relationships")
+
+	// Save data into Terraform state
+	resp.Diagnostics.Append(resp.State.Set(ctx, &data)...)
+}
+
+func (r *RelationshipResource) Read(ctx context.Context, req resource.ReadRequest, resp *resource.ReadResponse) {
+	var data RelationsipResourceModel
+
+	// Read Terraform prior state data into the model
+	resp.Diagnostics.Append(req.State.Get(ctx, &data)...)
+
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	rel, err := parseRelationship(data.Relationship.ValueString())
+	if err != nil {
+		resp.Diagnostics.AddError("Invalid Relationship", fmt.Sprintf("Unable to parse relationship: %s", err))
+		return
+	}
+
+	found, err := r.hasRelationship(ctx, rel)
+	if err != nil {
+		resp.Diagnostics.AddError("Unable to get relationship information", fmt.Sprintf("Unable to get relationship: %s", err))
+		return
+	}
+	if !found {
+		resp.State.RemoveResource(ctx)
+		return
+	}
+
+	tflog.Trace(ctx, "Read SpiceDB relationship")
+
+	resp.Diagnostics.Append(resp.State.Set(ctx, &data)...)
+}
+
+func (r *RelationshipResource) Update(ctx context.Context, req resource.UpdateRequest, resp *resource.UpdateResponse) {
```

Comment:
I am not sure if having `Update` not implemented would work?
Is it possible that in case of a change TF would think that SpiceDB was updated (because function returns) while no actual changes are performed?